oshi-core/src/main/java/oshi/hardware/platform/unix/solaris/SolarisComputerSystem.java

```diff
@@ -119,70 +141,48 @@ private static SmbiosStrings readSmbios() {
-        // Only works with root permissions but it's all we've got
+        SmbType smbTypeId = null;
+
+        EnumMap<SmbType, Map<String, String>> smbTypesMap = new EnumMap<>(SmbType.class);
+        smbTypesMap.put(SmbType.SMB_TYPE_BIOS, new HashMap<>());
+        smbTypesMap.put(SmbType.SMB_TYPE_SYSTEM, new HashMap<>());
+        smbTypesMap.put(SmbType.SMB_TYPE_BASEBOARD, new HashMap<>());
+
+//        // Only works with root permissions but it's all we've got
         for (final String checkLine : ExecutingCommand.runNative("smbios")) {
             // Change the smbTypeId when hitting a new header
-            if (checkLine.contains("SMB_TYPE_") && (smbTypeId = getSmbType(checkLine)) == Integer.MAX_VALUE) {
+            if (checkLine.contains("SMB_TYPE_") && (smbTypeId = getSmbType(checkLine)) == null) {
                 // If we get past what we need, stop iterating
                 break;
             }
             // Based on the smbTypeID we are processing for
-            switch (smbTypeId) {
-            case 0: // BIOS
-                if (checkLine.contains(vendorMarker)) {
-                    biosVendor = checkLine.split(vendorMarker)[1].trim();
-                } else if (checkLine.contains(biosVersionMarker)) {
-                    biosVersion = checkLine.split(biosVersionMarker)[1].trim();
-                } else if (checkLine.contains(biosDateMarker)) {
-                    biosDate = checkLine.split(biosDateMarker)[1].trim();
-                }
-                break;
-            case 1: // SYSTEM
-                if (checkLine.contains(manufacturerMarker)) {
-                    manufacturer = checkLine.split(manufacturerMarker)[1].trim();
-                } else if (checkLine.contains(productMarker)) {
-                    model = checkLine.split(productMarker)[1].trim();
-                } else if (checkLine.contains(serialNumMarker)) {
-                    serialNumber = checkLine.split(serialNumMarker)[1].trim();
-                } else if (checkLine.contains(uuidMarker)) {
-                    uuid = checkLine.split(uuidMarker)[1].trim();
-                }
-                break;
-            case 2: // BASEBOARD
-                if (checkLine.contains(manufacturerMarker)) {
-                    boardManufacturer = checkLine.split(manufacturerMarker)[1].trim();
-                } else if (checkLine.contains(productMarker)) {
-                    boardModel = checkLine.split(productMarker)[1].trim();
-                } else if (checkLine.contains(versionMarker)) {
-                    boardVersion = checkLine.split(versionMarker)[1].trim();
-                } else if (checkLine.contains(serialNumMarker)) {
-                    boardSerialNumber = checkLine.split(serialNumMarker)[1].trim();
-                }
-                break;
-            default:
-                break;
+            if (smbTypeId != null) {
+                String key = checkLine.substring(0, checkLine.indexOf(":")).trim();
```

Comment:
I think you need to test whether `indexOf(":")` is non-negative.

Bonus: you can capture the value in a variable and re-use it on these two lines.

Also I did look up whether string vs. character is better for `indexOf()`. The answer is it's a compiler intrinsic and both are replaced with identical code at compile time so it doesn't matter.